### PR TITLE
refactor/clang tidy/remove redundant casts

### PR DESCRIPTION
**Pull request by dundargoc**
- ci(clint): remove "Unknown NOLINT error category" warning
- refactor: remove redundant casts and disable false clang-tidy warnings

### CODE REVIEW

File: src/nvim/regexp.c [hunk 1]
```diff
@@ -5043,7 +5043,7 @@ static bool regmatch(
             } else {
               MB_PTR_BACK(rex.line, rp->rs_un.regsave.rs_u.ptr);
               if (limit > 0
-                  && (long)(behind_pos.rs_u.ptr
+                  && (long)(behind_pos.rs_u.ptr  // NOLINT(google-readability-casting)
                             - rp->rs_un.regsave.rs_u.ptr) > limit) {
```
Comment (bfredl) [inline src/nvim/regexp.c:5047]: perhaps we should cast limit to `ptrdiff_t` instead? (semantically preferred as `sizeof(ptr) >= sizeof(long)` )

File: src/nvim/ex_cmds2.c [hunk 2]
```diff
@@ -1796,7 +1796,7 @@ scriptitem_T *new_script_item(char_u *const name, scid_T *const sid_out)
   if (sid_out != NULL) {
     *sid_out = sid;
   }
-  ga_grow(&script_items, (int)(sid - script_items.ga_len));
+  ga_grow(&script_items, (int)(sid - script_items.ga_len));  // NOLINT(google-readability-casting)
```
Comment (bfredl) [inline src/nvim/ex_cmds2.c:1799]: remove cast instead. if type of sid or ga_len change later, we can re-add cast if needed.

File: src/nvim/eval/funcs.c [hunk 3]
```diff
@@ -7494,17 +7494,17 @@ static void f_readfile(typval_T *argvars, typval_T *rettv, FunPtr fptr)
          * small, to avoid repeatedly 'allocing' large and
          * 'reallocing' small. */
         if (prevsize == 0) {
-          prevsize = (long)(p - start);
+          prevsize = (long)(p - start);  // NOLINT(google-readability-casting)
```
Comment (bfredl) [inline src/nvim/eval/funcs.c:7497]: urgl, this is also a (pattern of) valid cast which shouldn't be an error.. (though maybe the real issue is that we should get rid of `long` eventually, it is an useless type which is size of ptr everywhere except for 64-bit MSVC)

Comment (jamessan) [reply src/nvim/eval/funcs.c:7497]: > (though maybe the real issue is that we should get rid of long eventually, it is an useless type which is size of ptr everywhere except for 64-bit MSVC)

And Linux x32

Comment (dundargoc) [reply src/nvim/eval/funcs.c:7497]: Would it be too janky to have a macro that only casts when it needs to? Something like (pseudocode):

```c
#ifdef WIN32
  #define CAST_TO_LONG (long)
#else
  #define CAST_TO_LONG
#endif
```

Alternatively I'm fine with just reverting this and the ones similar to this one, there are so few of them that it's not a big deal IMO.

Comment (bfredl) [reply src/nvim/eval/funcs.c:7497]: that's.. even worse. If just reverting this is okay, that would be better :]

**Comment by dundargoc** [comment]
CI passed let's goooooooooo!

**Pull request merged**